### PR TITLE
refactor: cleanup credential exchange code

### PR DESCRIPTION
**Pull request by Harasz**
### Description

Validate exchange credential types

### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or updated existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

File: src/modules/verifiable-credentials/verifiable-credentials-base.service.ts [hunk 1]
```diff
@@ -702,4 +697,22 @@ export abstract class VerifiableCredentialsServiceBase {
...
+   * This could be related to these issues:
+   *    - https://github.com/Sphereon-Opensource/pex/issues/96
+   *    - https://github.com/Sphereon-Opensource/pex/issues/91
+   * We are therefore trying to simplify the input to pex so remove the possibility of it generating incorrect results.
+   * Once the above issues are fixed, pex can be updated and perhaps this filtering will not needed.
+   *
+   * @param {InputDescriptorV1 | InputDescriptorV2} descriptors input descriptors
+   * @returns filtered input descriptors
+   */
+  private filterSelfSignDescriptors(
```
Comment (jrhender) [inline src/modules/verifiable-credentials/verifiable-credentials-base.service.ts:713]: @Harasz I guess your view is that `private` methods should be at the end of file? Is that what is motivating this change?

Comment (Harasz) [reply src/modules/verifiable-credentials/verifiable-credentials-base.service.ts:713]: yes, we are following that rule in other modules

Comment (jrhender) [reply src/modules/verifiable-credentials/verifiable-credentials-base.service.ts:713]: I agree with this convention. For me, this convention is coming from my experience with the C# ecosystem https://stackoverflow.com/a/310967 . Do you know where you got this convention from?

Comment (Harasz) [reply src/modules/verifiable-credentials/verifiable-credentials-base.service.ts:713]: you forced it somewhere and i liked this approach so you know :smile:

**Review by jrhender — APPROVED** [review]
Looks good to me

**Pull request merged**